libraries/AP_Scripting/applets/param-set.lua

```diff
@@ -0,0 +1,165 @@
+parameters_which_can_be_set["BATT_LOW_MAH"] = true
+parameters_which_can_be_set["BATT_LOW_VOLT"] = true
+parameters_which_can_be_set["BRD_OPTIONS"] = true
+parameters_which_can_be_set["COMPASS_USE3"] = true
+parameters_which_can_be_set["FENCE_ACTION"] = true
+parameters_which_can_be_set["FENCE_ALT_MAX"] = true
+parameters_which_can_be_set["FENCE_ENABLE"] = true
+parameters_which_can_be_set["FENCE_RADIUS"] = true
+parameters_which_can_be_set["FENCE_TYPE"] = true
+parameters_which_can_be_set["LIGHTS_ON"] = true
+parameters_which_can_be_set["LOG_BITMASK"] = true
+parameters_which_can_be_set["LOG_DISARMED"] = true
+parameters_which_can_be_set["LOG_FILE_DSRMROT"] = true
+parameters_which_can_be_set["RTL_ALT"] = true
+parameters_which_can_be_set["RTL_LOIT_TIME"] = true
+parameters_which_can_be_set["RTL_SPEED"] = true
+
+local function should_set_parameter_id(param_id)
+    if parameters_which_can_be_set[param_id] == nil then
+        return false
+    end
+    return true
+end
+
+local function handle_param_set(name, value)
+    gcs:send_text(MAV_SEVERITY.WARNING, string.format("%s: param set received", TEXT_PREFIX_STR))
+
+    -- we will not receive packets in here for the wrong system ID /
+    --   component ID; this is handled by ArduPilot's MAVLink routing
+    --   code
+
+    -- check for this specific ID:
+    if not should_set_parameter_id(name) then
+        gcs:send_text(MAV_SEVERITY.WARNING, string.format("%s: param set denied (%s)", TEXT_PREFIX_STR, name))
+        return
+    end
+
+    param:set_and_save(name, value)
+end
+
+-- display welcome message
+gcs:send_text(MAV_SEVERITY.INFO, "param-set script loaded")
+
+-- initialise our knowledge of the GCS's allow-set-parameters state.
+--   We do not want to fight over setting this GCS state via other
+--   mechanisms (eg. an auxiliary function), so we keep this state
+--   around to track what we last set:
+local gcs_allow_set = gcs:get_allow_param_set()
+
+-- update function to receive param_set messages and perhaps act on them
+local function update()
+    -- return immediately if not enabled
+    if (PARAM_SET_ENABLE:get() <= 0) then
+        -- this script is disabled, set allow-via-GCS (once):
+        if not gcs_allow_set then
+          gcs:set_allow_param_set(true)
+          gcs_allow_set = true
+        end
+        return
+    end
+
+    -- this script is enabled, disallow setting via normal means (once):
+    if gcs_allow_set then
+        gcs:set_allow_param_set(false)
+        gcs_allow_set = false
+    end
+
+    -- consume all available mavlink messages
+    while true do
+        local msg, _ = mavlink:receive_chan()
+        if msg == nil then
+            break
+        end
+
+        local param_value, target_system, target_component, param_id, _ = string.unpack("<fBBc16B", string.sub(msg, 13, 36))
+        param_id = string.gsub(param_id, string.char(0), "")
+
+        handle_param_set(target_system, target_component, param_id, param_value)
```

Comment:
```suggestion
        handle_param_set(param_id, param_value)
```

Reply:
That will probably result in unused variable warnings, so you will have to add underscores.

Reply:
Done, thanks.  How on earth did this pass tests, I wonder?!

Reply:
Didn't pass the scripting checks ;)